### PR TITLE
Space side diff

### LINKED ISSUE #168: Jump to corresponding diff when clicking on the chunk header

### LINKED ISSUE #178: Reposition canvas blocks on resize

### LINKED ISSUE #184: Width of space is not bounded to screen dimension
On small screens, it does not look good. https://github.com/preview-code/rite-evaluation/pull/182#issuecomment-222646179

### PR DESCRIPTION
**Pull request by TimvdLippe**
The space is now placed next to the diff view. When you click on the name of a block, you scroll to the corresponding diff on the right.

You can best test this PR with https://localhost:5200/projects/Polymer/Polymer/pulls/3680/diff

image

Fixes #168 
Fixes #178 
Fixes #184

### CODE REVIEW

**Comment by eanker** [comment]
image

As you already said, this doesn't work on small screens.

**Comment by TimvdLippe** [comment]
It should be a lot better now on your screen :)

File: src/projects/pull-request/pull-request-page.html [hunk 1]
```diff
@@ -114,6 +113,11 @@
         margin: 0 auto;
         width: 75%;
       }
+      iron-pages[wide] {
```
Comment (txsmith) [inline src/projects/pull-request/pull-request-page.html]: I think we need a different solution for this `wide` mode. At this point, when I click the 'Show Space' button, the entire pull-request-view goes into wide-mode, making the comments and description section way to wide. I want the wide-mode to only be active if the space view is visible.

File: src/projects/pull-request/pull-request-page.html [hunk 2]
```diff
@@ -159,22 +163,20 @@
...
       <pull-request-general-information data-route="overview" class="overview"
         pull-request="[[selectedPull]]" project="[[project]]"></pull-request-general-information>
       <pull-request-commits data-route="commits" pull-request="[[selectedPull]]" project="[[project]]"></pull-request-commits>
-      <pull-request-diff data-route="diff" diff="[[diff]]" ordering="[[ordering]]"
-          ordered-groups="{{orderedGroups}}"
-          base-url="[[selectedPull.head.repo.html_url]]" sha="[[selectedPull.head.sha]]"></pull-request-diff>
-      <pull-request-space data-route="space" groups="[[orderedGroups]]"></pull-request-space>
+      <pull-request-changes data-route="diff" diff="[[diff]]" ordering="[[ordering]]"
+          base-url="[[selectedPull.head.repo.html_url]]" sha="[[selectedPull.head.sha]]"
+          show-space="{{spaceShown}}"></pull-request-changes>
```
Comment (txsmith) [inline src/projects/pull-request/pull-request-page.html]: Maybe property should be set to false as soon as the space not _visible_

**Comment by txsmith** [comment]
When there is one chunk, and I click the `Link` button the linking-mode cannot be canceled, so I'll always see an arrow following the mouse pointer. Maybe a keybind on `ESC`?

File: src/projects/pull-request/pull-request-page.html [hunk 3]
```diff
@@ -114,6 +113,11 @@
         margin: 0 auto;
         width: 75%;
       }
+      iron-pages[wide] {
+        max-width: 1800px;
+        width: 95%;
+        margin: 0 20px;
```
Comment (txsmith) [inline src/projects/pull-request/pull-request-page.html]: This margin makes the element render out of the center

**Comment by TimvdLippe** [comment]
You can now hit escape and the button text changes to indicate you can reclick to stop the creation of a connection :tada:

**Comment by eanker** [comment]
It still does not work on my 13.3 inch screen.

image

This is with the lists collapsed.

**Comment by TimvdLippe** [comment]
Oops that was actually regressed while I was fixing the width calculations. It should be fixed now!

**Comment by txsmith** [comment]
Can the space and diff views be centered? It does not look very good on desktop-sized screens
screenshot from 2016-06-06 10 35 34

**Comment by txsmith** [comment]
When one navigation-list is shown, it slides over the space view. Is this intended?

**Comment by txsmith** [comment]
Scrolling with space-view open in this PR is broken: https://localhost:5200/projects/Polymer/Polymer/pulls/3680/diff

**Comment by txsmith** [comment]
We'll do more fixes in another PR.

**Pull request merged**